src/main/java/com/apuope/apuope_re/services/MyUserDetailService.java

```diff
@@ -0,0 +1,31 @@
+package com.apuope.apuope_re.services;
+
+import com.apuope.apuope_re.jooq.tables.records.UsersRecord;
+import com.apuope.apuope_re.model.UserPrinciple;
+import com.apuope.apuope_re.repositories.UserRepository;
+import org.jooq.DSLContext;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class MyUserDetailService implements UserDetailsService {
```

Comment:
To make the name fit in the context, could this be renamed e.g. ApuopeUserDetailService?